velox/functions/prestosql/Sequence.cpp

```diff
@@ -77,12 +77,11 @@ class SequenceFunction : public exec::VectorFunction {
     auto rawSizes = sizes->asMutable<vector_size_t>();
     auto rawOffsets = offsets->asMutable<vector_size_t>();
 
+    const bool isDate = args[0]->type() == DATE();
```

Comment:
does DATE() here construct a shared ptr? I wonder if we shall compare kind instead